packages/alchemy/src/middleware/gas-manager.ts

```diff
@@ -112,13 +112,16 @@ const withAlchemyGasAndPaymasterAndDataMiddleware = (
       await resolveProperties(struct)
     );
 
-    let feeOverride = undefined;
-    if (userOperation.maxFeePerGas && BigInt(userOperation.maxFeePerGas) > 0n) {
-      feeOverride = {
-        maxFeePerGas: userOperation.maxFeePerGas,
-        maxPriorityFeePerGas: userOperation.maxPriorityFeePerGas,
-      };
-    }
+    const overrides =
+      userOperation.maxFeePerGas && BigInt(userOperation.maxFeePerGas) > 0n
+        ? {
+            maxFeePerGas: userOperation.maxFeePerGas,
+            maxPriorityFeePerGas: userOperation.maxPriorityFeePerGas,
+            callGasLimit: userOperation.callGasLimit,
+            verificationGasLimit: userOperation.verificationGasLimit,
+            preVerificationGas: userOperation.preVerificationGas,
```

Comment:
overrides should be applied on a per field basis using similar logic: if the UO has the field set to a value > 0 then override the field otherwise leave as undefined

Reply:
these values here should be already overrides applied because we apply the overrides before the paymaster and data